project/src/main/java/com/google/sps/servlets/QueryServlet.java

```diff
@@ -70,6 +83,10 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
               .setLocation(getLatLngFromString(request.getParameter("location")))
               .setCuisines(ImmutableList.copyOf(request.getParameter("cuisines").split(",")))
               .build();
+      String userIdToken = request.getParameter("idToken");
+      if (!userIdToken.isEmpty()) {
```

Comment:
You can move this condition to storePreferences.

Reply:
Good idea